src/main/java/client/MyClient.java

```diff
@@ -1,117 +1,288 @@
+        myMessagePanel.add(myMessage);
+        myMessagePanel.add(send);
+        // добавление панелей на форму
+        add(panelmain);
+        add(myMessagePanel);
+        //add(jPanel);
+    }
+
+    private void saveMessage(JTextField myMessage) {
+        //String fullPATH = PATH.concat("history_").concat(loginPath).concat(".txt");
+        String fullPATH = "history_".concat(loginPath).concat(".txt");
+        serverService.saveMessage(myMessage.getText(),fullPATH);
+        myMessage.setText("");
+    }
+
+    private void sendMessage(JTextField myMessage) {
+        serverService.sendMessage(myMessage.getText());
+    }
+
+    private void loadHistory() {
+        //String fullPATH = PATH.concat("history_").concat(loginPath).concat(".txt");
+        String fullPATH = "history_".concat(loginPath).concat(".txt");
+        // получаем только последние 100 строк из истории
+        List<String> listHistory = serverService.loadHistory(fullPATH,100);
+        StringBuilder sbHistory = new StringBuilder();
+
+        Iterator iterator = listHistory.listIterator();
+
+        while (iterator.hasNext()){
+            sbHistory.append(iterator.next() + "\r\n");
         }
 
-        add(mainChat);
-        jPanel.add(send);
-        jPanel.add(myMessage);
-        add(jPanel);
+        String history = sbHistory.toString();
+        history.replace("\r\n","<br/>");
+
+        textFieldHistory.setText(history);
     }
 
     private void initLoginPanel(JTextArea mainChat) {
+
+        JPanel panel = new JPanel();
+        JPanel panelLogin = new JPanel();
+        JPanel panelButton = new JPanel();
+
+        panelLogin.setLayout(new BoxLayout(panelLogin, BoxLayout.Y_AXIS));
+        panelLogin.setSize(300, 50);
+        panelButton.setLayout(new BoxLayout(panelButton, BoxLayout.X_AXIS));
+        panelButton.setSize(300, 50);
+        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
+        panel.setSize(300, 50);
+
         JTextField login = new JTextField();
         login.setToolTipText("Логин");
         JPasswordField password = new JPasswordField();
         password.setToolTipText("Пароль");
+        JButton regButton = new JButton("Регистрация");
         JButton authButton = new JButton("Авторизоваться");
-
+        JButton swapButton = new JButton("Сменить ник");
+        JButton quitButton = new JButton("Выйти");
         JLabel authLabel = new JLabel("Offline");
-        authButton.addActionListener(actionEvent -> {
+
+        // регистрация
+        regButton.addActionListener(actionEvent -> {
             String lgn = login.getText();
             String psw = new String(password.getPassword());
+
             if (lgn != null && psw != null && !lgn.isEmpty() && !psw.isEmpty()) {
+                Registration_Authorization reg = new Registration_Authorization(lgn, psw);
+                reg.registrationUsers();
+                userOnline = reg.isRegistration();
+
+                if (reg.getMessageUser() != "") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.